ot/mpz-ot-core/src/kos/receiver.rs

```
@@ -469,6 +474,59 @@ impl ReceiverKeys {
             .map(|((key, c), ct)| if c { key ^ ct[1] } else { key ^ ct[0] })
             .collect())
     }
+
+    /// Decrypts the sender's payload.
+    ///
+    /// # Verifiable OT
+    ///
+    /// Verifiable OT with KOS does not currently support byte payloads, so no record of this payload
+    /// will be recorded.
+    pub fn decrypt_bytes<const N: usize>(
+        self,
+        payload: SenderPayload,
+    ) -> Result<Vec<[u8; N]>, ReceiverError> {
+        let SenderPayload { id, ciphertexts } = payload;
+
+        let Ciphertexts::Bytes { ciphertexts, iv, length  } = ciphertexts else {
+            return Err(ReceiverError::InvalidPayload);
+        };
+
+        if id != self.id {
+            return Err(ReceiverError::IdMismatch(self.id, id));
+        }
+
+        let length = length as usize;
+        if ciphertexts.len() / (2 * length) != self.keys.len() {
+            return Err(ReceiverError::CountMismatch(
+                self.keys.len(),
+                ciphertexts.len() / (2 * length),
+            ));
+        }
+
+        let iv: [u8; 16] = iv.try_into().map_err(|_| ReceiverError::InvalidPayload)?;
```

This is a CTR nonce which must be only 8 bytes. I don't understand why `Aes128Ctr::new` below silently accepts it without giving an error.

---

It is an IV, which gets combined with the CTR nonce. The CTR nonce is per block, the IV is per message